src/lib/operators/export_binary.cpp

```diff
@@ -171,95 +149,75 @@ void ExportBinary::_write_header(const Table& table, std::ofstream& ofstream) {
-template <>
-void ExportBinary::ExportBinaryVisitor<pmr_string>::handle_segment(
-    const ReferenceSegment& ref_segment, std::shared_ptr<SegmentVisitorContext> base_context) {
-  auto context = std::static_pointer_cast<ExportContext>(base_context);
+  if (reference_segment.data_type() == DataType::String) {
+    std::stringstream values;
+    pmr_string value;
+    pmr_vector<size_t> string_lengths(reference_segment.size());
 
-  // We materialize reference segments and save them as value segments
-  export_value(context->ofstream, BinarySegmentType::value_segment);
-
-  // If there is no data, we can skip all of the coming steps.
-  if (ref_segment.size() == 0) return;
+    // We export the values materialized
+    for (ChunkOffset row = 0; row < reference_segment.size(); ++row) {
+      value = boost::get<pmr_string>(reference_segment[row]);
+      string_lengths[row] = value.length();
+      values << value;
+    }
 
-  std::stringstream values;
-  pmr_string value;
-  std::vector<size_t> string_lengths(ref_segment.size());
+    export_values(ofstream, string_lengths);
+    ofstream << values.rdbuf();
 
-  // We export the values materialized
-  for (ChunkOffset row = 0; row < ref_segment.size(); ++row) {
-    value = boost::get<pmr_string>(ref_segment[row]);
-    string_lengths[row] = value.length();
-    values << value;
+  } else {
+    resolve_data_type(reference_segment.data_type(), [&](auto type) {
+      using SegmentDataType = typename decltype(type)::type;
+      // Unfortunately, we have to iterate over all values of the reference segment
+      // to materialize its contents. Then we can write them to the file
+      for (ChunkOffset row = 0; row < reference_segment.size(); ++row) {
+        export_value(ofstream, boost::get<SegmentDataType>(reference_segment[row]));
+      }
+    });
   }
-
-  export_values(context->ofstream, string_lengths);
-  context->ofstream << values.rdbuf();
 }
 
-template <typename T>
-void ExportBinary::ExportBinaryVisitor<T>::handle_segment(const BaseDictionarySegment& base_segment,
-                                                          std::shared_ptr<SegmentVisitorContext> base_context) {
-  auto context = std::static_pointer_cast<ExportContext>(base_context);
-
-  Assert(base_segment.compressed_vector_type(),
+void ExportBinary::_write_segment(const BaseDictionarySegment& base_dictionary_segment, std::ofstream& ofstream) {
+  Assert(base_dictionary_segment.compressed_vector_type(),
          "Expected DictionarySegment to use vector compression for attribute vector");
-  Assert(is_fixed_size_byte_aligned(*base_segment.compressed_vector_type()),
+  Assert(is_fixed_size_byte_aligned(*base_dictionary_segment.compressed_vector_type()),
          "Does only support fixed-size byte-aligned compressed attribute vectors.");
-
-  export_value(context->ofstream, BinarySegmentType::dictionary_segment);
+  export_value(ofstream, EncodingType::Dictionary);
```

Comment:
This means that we convert FixedString dictionaries to regular dictionaries. That is probably surprising for the caller. As we don't have an import for the former yet, I'd change the support from BaseDictionarySegment to DictionarySegment.